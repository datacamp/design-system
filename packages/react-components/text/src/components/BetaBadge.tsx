import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

import Badge from './Badge';

interface BetaBadgeProps {
  /**
   * Additional css className to add to the rendered element
   */
  className?: string;
}

const betaBadgeStyle = css({
  background: tokens.color.opaque.purple.value.hex,
  color: 'white',
});

const BetaBadge = ({ className }: BetaBadgeProps): React.ReactElement => (
  // color is overriden by betaBadgeStyle
  <Badge className={className} color="" css={betaBadgeStyle} size="large">
    Beta
  </Badge>
);

export default BetaBadge;
