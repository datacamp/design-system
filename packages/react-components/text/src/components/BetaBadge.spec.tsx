import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import React from 'react';

import BetaBadge from './BetaBadge';

describe('BetaBadge', () => {
  it('renders with the provided className', async () => {
    const { getByText } = await axeRender(<BetaBadge className="test-class" />);
    const element = getByText('Beta') as HTMLElement;
    expect(element).toHaveClass('test-class');
    expect(element).toMatchSnapshot();
  });
});
