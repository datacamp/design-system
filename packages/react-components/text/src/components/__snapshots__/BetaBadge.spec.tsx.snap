// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetaBadge renders with the provided className 1`] = `
.emotion-0 {
  -webkit-font-smoothing: antialiased;
  color: rgb(77,83,86);
  font-family: sans-serif;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  font-weight: 350;
  line-height: 16px;
  border-radius: 4px;
  display: inline-block;
  text-transform: uppercase;
  color: white;
  font-size: 16px;
  line-height: 24px;
  padding-left: 8px;
  padding-right: 8px;
  background: linear-gradient(-136deg,#c06ac8,#8468c4);
  color: white;
}

<strong
  class="test-class emotion-0"
>
  Beta
</strong>
`;
